airflow/jobs/scheduler_job.py

```diff
@@ -316,6 +316,15 @@ def _executable_task_instances_to_queued(self, max_tis: int, session: Session =
 
         pool_to_task_instances: DefaultDict[str, List[models.Pool]] = defaultdict(list)
         for task_instance in task_instances_to_examine:
+            # If the dag is no longer in the dagbag, don't bother
```

Comment:
How is that case possible?

The path that leads to here is:
`_do_scheduling` -> `_critical_section_execute_task_instances` -> `_executable_task_instances_to_queued`

It's odd that dagruns were created but the task eventually is "dagless". How was dagrun created if there is no dag in the dag bag?
Is this designed to handle the edge case of removing the DAG after a dagrun was created?

Reply:
This can hardly happens but can happen.
The way to reproduce it is to remove a dag file of a running dag and mark the task instances as failed then delete the dag from the UI. You'll observe the crashlooping

As for the change, it doesn't stop a running task from completing, it only errors out that the dag is missing. 
I could actually only error when checking concurrency because that's where we accessed a dagbag and used it to check for task which can cause the scheduler to crash with nonetype attribute error.

And sorry for the misleading comment at the concurrency point. The task would get there and still print error log.

I'm considering to only have the log there. WDYT?